src/SubGraphFunctions.jl

```diff
@@ -128,7 +128,7 @@ function transferUpdateSubGraph!(dest::AbstractDFG,
   # transfer specific fields into dest from src
   for var in (x->getVariable(src, x)).(syms)
     # copy not required since a broadcast is used internally
-    updateVariableSolverData!(dest, var, solveKey, false, [:val; :bw; :inferdim; :solvedCount]; warn_if_absent=false)
+    updateVariableSolverData!(dest, var, solveKey, false, [:val; :bw; :inferdim; :solvedCount; :initialized]; warn_if_absent=false)
```

Comment:
with tree init, :initialized is set in subfg but not updated in full fg. There might be other flags also needed.